### DISCUSSION TITLE
How run SQLPage on Synology NAS with distant access ?

### DISCUSSION THREAD

**Discussion by ThierryM** [sqlpage/SQLPage#267] [Q&A] [answered]
Hi,
Thanks a lot for your work and your sharing.
I've installed SQLPage on my Synology NAS (DSM 7.2) with docker and it's running locally without problem with this address 192.168.1.30000 : 
Synology NAS 192.168.1.32:30000 --> SQLPage Docker port 8080

But I want to access SQLPage from Internet. It's possible because Synology provide a reverse proxy that redirect the external address like 'sqlpage.name.synology.me" to an local adress on my server with the port 30000 redirecting to the docker port 8080 of SQLPage.
This method works for other dockerised applications but not for SQLPage.
I tried to add a sqlpage.json on /etc/sqlpage folder mounted but with no succeeding with the option "https_domain".
Here a schema of what I would do : 
external access with the address "sqlpage.name.synology.me" --> Home router --> Synology NAS : 192.168.1.32:30000 --> SQLPage Docker port 8080
So if you 've an idea ;-)... Thanks for your help.
Regards,

Thierry

**Comment by lovasoa** [comment]
Hello ! I have never used a synology NAS, but I don't see any reason this shouldn't work. There does not seem to be anything SQLPage-specific here. You should check the configuration, and if everything seems to be in order, ask on the https://community.synology.com/enu synology community forums :slightly_smiling_face: 

You shouldn't have to use any special configuration in SQLPage, since the https connection is going to be handled by synology. So you just serve a vanilla SQLPage, without https, and let the reverse proxy terminate the https connection.

The setup shouldlook like this `client <----[https]-----> reverse proxy <----[ http ]----> SQLPage`

https://kb.synology.com/en-af/DSM/help/DSM/AdminCenter/system_login_portal_advanced?version=7

**Reply by ThierryM** [reply]
Thanks a lot for your answer.
I've checked all but it is still not working. If I find the solution, I'll post here what was the problem.
Regards,
Thierry

**Comment by ThierryM** [comment]
Hi,
~~I found the solution : it's necessary to add a Websocket header in the reverse proxy configuration :~~
![Sélection_564](https://github.com/lovasoa/SQLpage/assets/1470067/374cde82-b993-4ce2-9f51-bbea018c1849)
Regards.

**Reply by lovasoa** [reply]
That seems strange: SQLPage doesn't use websockets...

**Reply by ThierryM** [reply]
Whitout that setting, when I want to access from outside to SQLPage, the Synology NAS return an error page. It's probably specific with NAS Synology.

**Comment by ThierryM** [comment, marked as answer]
I found an other explanation for this problem.
In the reverse proxy settings, if I put the real IP address of the NAs server, (192.168.1.xx) I've the problem.
But if I put "localhost", the problem disappears. So it's not a WebSocket problem. Sorry for the noise...
![Sélection_565](https://github.com/lovasoa/SQLpage/assets/1470067/806024e6-90b8-4a5f-98a5-8d1427ee6647)
Regards

**Reply by lovasoa** [reply]
Yes, that sounds like the real explanation. Thanks for sharing!
